tools/coredump/ebpfcode.go

```diff
@@ -12,8 +12,16 @@ package main
 // Also the tail call helper "bpf_tail_call" is overridden here, as it works in
 // tandem with the main entry point's setjmp.
 
+// todo: enable -Wunused-parameter after 1.25 upgrade https://go-review.googlesource.com/c/go/+/642196
+
 /*
-#define TESTING
+#cgo CFLAGS: -Wall -Wextra -Werror
+#cgo CFLAGS: -Wno-address-of-packed-member
+#cgo CFLAGS: -Wno-unused-label
+#cgo CFLAGS: -Wno-sign-compare
+#cgo CFLAGS: -Wno-unknown-pragmas
+#cgo CFLAGS: -Wno-unused-parameter
+#cgo CFLAGS: -Wno-strict-aliasing
```

Comment:
Not sure about this one as well.
```
In file included from ./ebpfcode.go:61:
./../../support/ebpf/hotspot_tracer.ebpf.c: In function 'hotspot_handle_epilogue':
./../../support/ebpf/hotspot_tracer.ebpf.c:537:10: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  537 |     if (*(u64 *)&window[find_offset] == needle) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```
will need to read and learn about this.

Reply:
It just says that here a u32 array is accessed via a u64 read (with a cast to a u64 pointer before). The aliasing (optimal or allowed memory boundaries) doesn't fit any more. For u64 it is 8 byte boundaries while for u32 it is 4 bytes boundaries.

You could make `window` an u64 array with half of the elements, and change the code appropriately. @fabled may know this code best.

Reply:
We can disable strict-aliasing `-fno-strict-aliasing`, if people are no aware of it and aliasing warnings are also disabled we can run into problems with the optimizer.